stratum/pipelines/stratum-tna/stratum_tna_test.cc

```diff
@@ -107,6 +107,37 @@ class StratumTnaTest : public P4RuntimeFixture {
     return route_entry;
   }
 
+  ::p4::v1::ActionProfileMember CreateActionProfileMember(uint32 member_id,uint16 dst_port, uint64 dst_mac) {
+    ::p4::v1::ActionProfileMember ap_entry =
+        HydrateP4RuntimeProtoFromStringOrDie<::p4::v1::ActionProfileMember>(
+            p4_id_replacements_,
+            R"PROTO(
+        action_profile_id: {StratumIngress.ipv4_ap}
+          action {
+            action_id: {StratumIngress.fwd_route}
+            params {
+              param_id: {StratumIngress.fwd_route.port}
+              value: "\x00"
+            }
+            params {
+              param_id: {StratumIngress.fwd_route.dmac}
+              value: "\x00"
+            }
+          }
+      )PROTO");
+      // TODO what is the method membership space here ???? (renner)
+    // ap_entry.mutable_member_id()->set_value(member_id);
```

Comment:
@pudelkoM is there an easy way to check type method membership for types in the cc generated code ? I wasnt able to untangle what the contents of an object like 
```
::p4::v1::ActionProfileMember ap_entry =
```
would be ?

Reply:
If your SDE supports auto-completion for generated code, you can cycle through it until you find what you need.
Reading the proto source also gives a good overview which fields are available: https://github.com/p4lang/p4runtime/blob/main/proto/p4/v1/p4runtime.proto#L286
Together with the [C++ generated code guide](https://developers.google.com/protocol-buffers/docs/reference/cpp-generated) I usually manage.

Reply:
forgot i always used to use the go generated code guide.... thanks for the reminder now see that in fact there is an exact set for that field thank you sir !!
```
$ ag set_member_id
p4runtime.pb.h
4500:  void set_member_id(::PROTOBUF_NAMESPACE_ID::uint32 value);
```